pkg/ccl/sqlproxyccl/connector.go

```diff
@@ -319,7 +319,11 @@ func (c *connector) lookupAddr(ctx context.Context) (string, error) {
 	if c.Directory != nil {
 		addr, err := c.Directory.EnsureTenantAddr(ctx, c.TenantID, c.ClusterName)
 		if err != nil {
-			if status.Code(err) != codes.NotFound {
+			if status.Code(err) == codes.FailedPrecondition {
+				if st, ok := status.FromError(err); ok {
```

Comment:
If we are unable to retrieve the status, we should still return a codeUnavailable error and break from the loop. I'm worried this case could trigger an infinite loop.